chain/chain/src/chain.rs

```diff
@@ -1359,12 +1359,20 @@ impl Chain {
         let Some((block, chunks)) = self.optimistic_block_chunks.take_latest_ready_block() else {
             return;
         };
-
-        self.blocks_delay_tracker.record_optimistic_block_ready(block.height());
-
         let prev_block_hash = *block.prev_block_hash();
         let block_hash = *block.hash();
         let block_height = block.height();
+
+        self.blocks_delay_tracker.record_optimistic_block_ready(block_height);
+        if let Ok(true) = self.is_height_processed(block_height) {
+            debug!(
+                target: "chain", prev_block_hash = ?prev_block_hash,
+                hash = ?block_hash, height = block_height,
+                "Dropping optimistic block, the height was already processed"
+            );
+            return;
```

Comment:
Do we care enough to add a metric?

Not *that* useful, but counters are relatively cheap